agent/agent.go

```diff
@@ -2070,52 +2119,39 @@ func (a *Agent) loadChecks(conf *Config) error {
 
 // unloadChecks will deregister all checks known to the local agent.
 func (a *Agent) unloadChecks() error {
-	for _, check := range a.state.Checks() {
-		if err := a.RemoveCheck(check.CheckID, false); err != nil {
-			return fmt.Errorf("Failed deregistering check '%s': %s", check.CheckID, err)
+	for id := range a.State.Checks() {
```

Comment:
I wonder if the unload for services and checks should be added as a method on the state, since this is technically working off a copy so this isn't atomic.